Zend/tests/function_arguments/sensitive_parameter_value.phpt

```diff
@@ -0,0 +1,18 @@
+--TEST--
+The SensitiveParameterValue behaves as expected.
+--FILE--
+<?php
+
+$v = new SensitiveParameterValue('secret');
+
+var_dump($v);
+var_dump(serialize($v));
+var_dump(json_encode($v));
+var_dump($v->getValue());
```

Comment:
Good catch. I believe it should: The only way of accessing the sensitive value should be the explicit call to `->getValue()`.

Is there an existing example of how the cast to array can be controlled?

Reply:
I don't know this code but it looks like the VM will call the `get_properties_for` handler and passing the `ZEND_PROP_PURPOSE_ARRAY_CAST` flag, which would allow you to handle the case in whatever way is appropriate.

https://github.com/php/php-src/blob/bf32299184b69b505681ee05d63872563a2c7591/Zend/zend_vm_def.h#L6201-L6212

Reply:
Thanks, implemented in f031d1688c47b54d8934e3dd9707a4b42c92d132.